src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java

```diff
@@ -0,0 +1,31 @@
+package org.opentripplanner.middleware.utils;
+
+import org.opentripplanner.middleware.otp.response.Itinerary;
+import org.opentripplanner.middleware.otp.response.Leg;
+
+public class ItineraryUtils {
+    /**
+     * Determines whether the specified {@link Itinerary} can be monitored.
+     * @return true if at least one {@link Leg} of the specified {@link Itinerary} is a transit leg,
+     *   and none of the legs is a rental or ride hail leg (e.g. CAR_RENT, CAR_HAIL, BICYCLE_RENT, etc.).
+     *   (We use the corresponding fields returned by OTP to get transit legs and rental/ride hail legs.)
+     */
+    public static boolean itineraryCanBeMonitored(Itinerary itinerary) {
+        boolean hasTransit = false;
+        boolean hasRentalOrRideHail = false;
+
+        if (itinerary != null && itinerary.legs != null) {
+            for (Leg leg : itinerary.legs) {
+                if (leg.transitLeg != null && leg.transitLeg) {
+                    hasTransit = true;
+                }
+                if (leg.rentedVehicle != null && leg.rentedVehicle ||
+                    leg.hailedCar != null && leg.hailedCar) {
```

Comment:
This is missing a check for `leg.rentedBike` and `leg.rentedCar`.

Reply:
Updated in https://github.com/ibi-group/otp-middleware/pull/91/commits/43425abecb53ee8d559c0d7c75062e6c84b7f03d.